### PR TITLE
Apply 'go mod tidy' after API sync

### PR DESCRIPTION
**Pull request by mmorel-35**
@phlax ,
I believe there is no go mod tidy applied after envoy synchronisation

### CODE REVIEW

File: ci/sync_envoy.sh [hunk 1]
```diff
@@ -72,5 +72,6 @@ commit_changes () {
 
 
 build_protos
+go mod tidy
```
Comment (phlax) [inline ci/sync_envoy.sh:75]: i think lets either put it to its own function or add it after the build

**Review by phlax — COMMENTED** [review]
dont worry about testing it we can land and if there is an issue we can revert

the only question i have is what go version this is using - one way/another i think we want to make sure its pinned

**Comment by phlax** [comment]
trying to rem how i tested previously - maybe i copied the workflow to trigger on pr altho cant rem if that is allowed for prs to do

**Comment by phlax** [comment]
not sure - im kinda surprised this repo uses make tbh

iiuc you would mostly want to run tidy after changing deps so probs a standalone target - but i have limited go experience, so happy to defer to others with more knowledge of go workflows

**Review by sunjayBhatia — APPROVED** [review]
I think what you have here is sufficient, a new make target would be a bit verbose for just running `go mod tidy` and `make clean` isn't necessary, that seems more for a local dev environment where you may have old binaries etc lying around

**Pull request merged**